src/Modules/Finance/Header/Header.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { TargetSum } from './TargetSum/TargetSum';
+import { Total } from './Total/Total';
+import { Transaction } from './Transaction/Transaction';
+
+import styles from './Header.module.css';
+
+type IHeaderProps = {
+
+}
+
+type IHeaderState = {
+	targetSumValue: string | number | any,
+	targetSumConfirmed: boolean,
+	totalSum: string,
+	currentTransactionValue: string
+}
+
+export class Header extends React.Component<IHeaderProps, IHeaderState> {
+
+	state: IHeaderState = {
+		targetSumValue: '',
+		targetSumConfirmed: false,
+		totalSum: '0',
+		currentTransactionValue: '0'
+	}
+
+	targetSumConfirmedHandler = () => {
+		this.setState({ targetSumConfirmed: true });
+	}
+
+	targetSumInputChangedHandler = (event: any) => {
+		this.setState({ targetSumValue: event.target.value });
+	}
+
+	render() {
+		return (
+			<div className={styles.Header}>

```

Comment:
Concerning markup: as for me I'd expect the layout for Header to be specified in the Header component itself. For now there is only wrapper and we have no idea what classes (and styles) will be given to the Header's children. 
I'd define the wrapper and containers for children in the Header. This way:
```
<headerWrapper>
    <headerChild>{ANY_COMPONENT}</headerChild>
    <headerChild>{ANY_COMPONENT}</headerChild>
</headerWrapper>
```
Why? 
1. You keep all the Header related styling in one place. In case you need to update styling it'll be easy.
2. Components, used as children, won't contain any header-specific styles, thus it'll be easier to reuse them. (I guess now you provide them with size-styles).

Reply:
You are right. It makes sence.